sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/highlighting/XMLHighlighting.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * SonarQube XML Plugin
+ * Copyright (C) 2010 SonarSource
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.xml.highlighting;
+
+import com.google.common.io.Files;
+
+import javax.xml.stream.Location;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+public class XMLHighlighting {
+
+  private static final String XML_DECLARATION_TAG = "<?xml";
+
+  private List<HighlightingData> highlighting = new ArrayList<>();
+  private String content;
+
+  public XMLHighlighting(File file, Charset charset) throws Exception{
+    content = Files.toString(file, charset);
+    highlightXML(new FileInputStream(file));
+  }
+
+  public XMLHighlighting(String xmlStrContent) throws XMLStreamException{
+    content = xmlStrContent;
+    InputStream inputStream = new ByteArrayInputStream(xmlStrContent.getBytes(StandardCharsets.UTF_8));
+    highlightXML(inputStream);
+  }
+
+  public List<HighlightingData> getHighlightingData() {
+    return highlighting;
+  }
+
+  public void highlightXML(InputStream inputStream) throws XMLStreamException {
+    XMLInputFactory factory = XMLInputFactory.newInstance();
+    factory.setProperty(XMLInputFactory.SUPPORT_DTD, "false");
+    XMLStreamReader xmlReader = factory.createXMLStreamReader(inputStream);
```

Comment:
@vilchik-elena This will probably use the default platform encoding instead of the encoding which was configured on the project. We should probably use the method which accepts a Reader instead of an InputStream.